test/Microsoft.AspNetCore.Razor.Language.Test/TestFiles/ParserTests/HtmlToCodeSwitchTest/DoesNotSwitchToCodeOnEmailAddressInAttribute.stree.txt

```diff
@@ -1,27 +1,31 @@
-Markup block - Gen<None> - 50 - (0:0,0)
-    Tag block - Gen<None> - 38 - (0:0,0)
-        Markup span - Gen<Markup> - [<a] - SpanEditHandler;Accepts:Any - (0:0,0) - Tokens:2
-            SyntaxKind.OpenAngle;[<];
-            SyntaxKind.Text;[a];
-        Markup block - Gen<Attr:href, href="@(2:0,2),"@(36:0,36)> - 35 - (2:0,2)
-            Markup span - Gen<None> - [ href="] - SpanEditHandler;Accepts:Any - (2:0,2) - Tokens:4
-                SyntaxKind.Whitespace;[ ];
-                SyntaxKind.Text;[href];
-                SyntaxKind.Equals;[=];
-                SyntaxKind.DoubleQuote;["];
-            Markup span - Gen<LitAttr:@(9:0,9)> - [mailto:anurse@microsoft.com] - SpanEditHandler;Accepts:Any - (9:0,9) - Tokens:1
-                SyntaxKind.Text;[mailto:anurse@microsoft.com];
-            Markup span - Gen<None> - ["] - SpanEditHandler;Accepts:Any - (36:0,36) - Tokens:1
-                SyntaxKind.DoubleQuote;["];
-        Markup span - Gen<Markup> - [>] - SpanEditHandler;Accepts:None - (37:0,37) - Tokens:1
-            SyntaxKind.CloseAngle;[>];
-    Markup span - Gen<Markup> - [Email me] - SpanEditHandler;Accepts:Any - (38:0,38) - Tokens:3
-        SyntaxKind.Text;[Email];
-        SyntaxKind.Whitespace;[ ];
-        SyntaxKind.Text;[me];
-    Tag block - Gen<None> - 4 - (46:0,46)
-        Markup span - Gen<Markup> - [</a>] - SpanEditHandler;Accepts:None - (46:0,46) - Tokens:4
-            SyntaxKind.OpenAngle;[<];
-            SyntaxKind.ForwardSlash;[/];
-            SyntaxKind.Text;[a];
-            SyntaxKind.CloseAngle;[>];
+MarkupBlock - [0..50)::50 - [<a href="mailto:anurse@microsoft.com">Email me</a>]
+    MarkupTagBlock - [0..38)::38 - [<a href="mailto:anurse@microsoft.com">]
+        MarkupTextLiteral - [0..2)::2 - [<a] - Gen<Markup> - SpanEditHandler;Accepts:Any
+            OpenAngle;[<];
+            Text;[a];
+        MarkupAttributeBlock - [2..37)::35 - [ href="mailto:anurse@microsoft.com"]
+            MarkupTextLiteral - [2..3)::1 - [ ] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                Whitespace;[ ];
+            MarkupTextLiteral - [3..7)::4 - [href] - Gen<Markup> - SpanEditHandler;Accepts:Any
+                Text;[href];
+            Equals;[=];
+            MarkupTextLiteral - [8..9)::1 - ["] - Gen<None> - SpanEditHandler;Accepts:Any
+                DoubleQuote;["];
+            GenericBlock - [9..36)::27
```

Comment:
Maybe informational for me - what is `GenericBlock`?

Reply:
I explained it in the previous PR. https://github.com/aspnet/Razor/pull/2609#discussion_r220375373. Pasting it below,

ajaybhargavb 18 hours ago  Member
> In case you are wondering, this is to address @NTaylorMullen's comment in earlier PR (What's the difference between HtmlBlock and HtmlMarkupBlock?). GenericBlock is basically a wrapper when we need some parent block. It doesn't have a syntactic significance.

NTaylorMullen 18 hours ago  Member
> Container?

ajaybhargavb 18 hours ago  Member
> Yes, that's the word